app/models/package_manager/maven.rb

```diff
@@ -17,27 +18,52 @@ class Maven < Base
       "http://www.eclipse.org/org/documents/edl-v10" => "Eclipse Distribution License (EDL), Version 1.0",
     }.freeze
 
+    PROVIDER_MAP = {
+      "SpringLibs" => SpringLibs,
+      "Maven" => MavenCentral,
+    }.freeze
+
     def self.package_link(project, version = nil)
+      if version
+        db_version = project.versions.find_by(number: version)
+        unless db_version&.repository_sources.blank?
+          repository_source = db_version.repository_sources.first
+          return PROVIDER_MAP[repository_source].package_link(project, version)
+        end
+      end
+
       MavenUrl.from_name(project.name, repository_base).search(version)
     end
 
     def self.download_url(name, version = nil)
+      if version
+        project = Project.find_by(name: name, platform: "Maven")
+        db_version = project.versions.find_by(number: version)
+        unless db_version&.repository_sources.blank?
+          repository_source = db_version.repository_sources.first
+          return PROVIDER_MAP[repository_source].download_url(name, version)
+        end
+      end
+
       MavenUrl.from_name(name, repository_base).jar(version)
     end
 
     def self.check_status_url(project)
-      MavenUrl.from_name(project.name, repository_base).base
-    end
+      sources = project.versions.flat_map(&:repository_sources).compact.uniq
+      return PROVIDER_MAP[sources.first].check_status_url(project) unless sources.blank?
 
-    def self.repository_base
-      "https://repo1.maven.org/maven2"
+      MavenUrl.from_name(project.name, repository_base).base
     end
 
     def self.load_names(limit = nil)
       names = get("https://maven.libraries.io/all")
       names.each { |name| REDIS.sadd("maven-names", name)}
     end
 
+    def self.repository_base
```

Comment:
Should this live in the maven provider?

Reply:
I still need to test what removing this stuff from here will do. Ideally it wouldn't live here.